### PR TITLE
Implement transaction center

### PR DESCRIPTION
**Pull request by MeBrei**
As part of #202 this component will show transactions and their states in an expandable list.

Depends on #235 

ToDos
- [ ] format txs with actual data
  - [x] tx message - needs fixing of `IntrospectionFragmentMatcher` error to retrieve the tx kind
  - [ ] tx status/progress - needs block heights to compare to the threshold (#225)
  - [ ] error message - to display the cause of a failed transaction (#225)
  - [ ] timestamp - we need timestamps that reflect settle and error events (#225)
- [x] add box shadow & error coloring
- [ ] add (or write issue on) animation
    - [ ] slide up/down cards
    - [ ] on first appearance of the tx center, slide it into view
- [x] Transaction detail view
  - [x] add correct icon design
  - [x] format timestamp
  - [x] retrieve user/project info
- [ ] move transaction query into a central storage

The TxAccordion component can be viewed in the `DesignSystemGuide` `shift-d` with mock data. The Tx center can be viewed by submitting a transaction (eg register your user `shift-t` - note that you have to create an identity first `shift-i`) and opening any view that uses the sidebar layout, as currently transactions are only queried on sidebar layout load.

### CODE REVIEW

**Comment by MeBrei** [comment]
### `IntrospectionFragmentMatcher` issue

The transaction `messages` use fragments with different fields. This causes an error with our current apollo implementation (even if we query the same fields for each fragment).
Screenshot 2020-03-26 at 13 52 35
Screenshot 2020-03-26 at 13 52 49

The link in the error does not work, but I found [another one](https://www.apollographql.com/docs/react/data/fragments/#fragments-on-unions-and-interfaces).

**Comment by MeBrei** [comment]
The 2 todos dependent on #225 can't be addressed right now.

I will create a followup issue for the animation todos as they also depend on how we poll the transactions continuously in the future.

For now only the tx message is using the actual data. The progress and status are hardcoded.

File: ui/DesignSystem/Component/TxAccordion.svelte [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  .hidden {
+    display: none;
+  }
+</style>
+
+<div class="pipeline" class:negative={iconState === 'negative'} {style}>
+  <div class="cards" class:hidden>
+    {#each transactions as tx}
+      <!-- TODO: Link card to tx detail view -->
+      <div class="card">
```
Comment (xla) [inline ui/DesignSystem/Component/TxAccordion.svelte]: This nesting is quite expansive, can we refactor this part out into a distinct card component?

Comment (MeBrei) [reply ui/DesignSystem/Component/TxAccordion.svelte]: In which way do you mean expensive?

I like the idea of having all the logic and styling of this component in one file. For now we also only need the `card` in this component so that I would tend to keeping it here. 
@rudolfs What's your opinion on this?

Comment (xla) [reply ui/DesignSystem/Component/TxAccordion.svelte]: Not expensive, wrote expansive, as in the nesting of the markup here is quite deep. It's not extraction for reusability but for readability. If there is no strong concern shared it's fine to leave it as is, this was the only place that gave me a stumble while reviewing.

Comment (rudolfs) [reply ui/DesignSystem/Component/TxAccordion.svelte]: > @rudolfs What's your opinion on this?

I'm fine with it either way. I wish there was a better way of handling `<Flex>` without needing to nest it twice, but I don't know of a better solution at the moment either.

**Comment by rudolfs** [comment]
> add (or write issue on) animation

Have a look at [svelte transitions](https://svelte.dev/docs#transition_fn). We already use them in some parts, like:
```
ui/DesignSystem/Component/AdditionalActionsDropdown.svelte
ui/DesignSystem/Component/NotificationFaucet.svelte
ui/DesignSystem/Component/RadioOption.svelte
```

**Comment by MeBrei** [comment]
I will add the transaction detail view in this PR as well. (Will dismiss the approvals)

File: ui/DesignSystem/Component/TxAccordion.svelte [hunk 2]
```diff
@@ -0,0 +1,171 @@
+<script>
+  import { Flex, Icon, Text } from "../Primitive";
+  // TODO: Make component responsive to transactions change
```
[the remark this answers is not in the document]
Comment (MeBrei) [reply ui/DesignSystem/Component/TxAccordion.svelte]: I will create a follow up issue to move the transactions into a central store. I will then also address this todo

**Comment by MeBrei** [comment]
In order to not let this PR grow too big, I moved some of the outstanding tasks outside of this PR and noted them in #202

File: ui/DesignSystem/Component/Transaction/Accordion.svelte [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  <div class="cards" class:hidden>
+    {#each transactions as tx}
+      <div
+        class="card"
+        on:click={() => {
+          push(path.transactions(tx.id));
+        }}>
+        <Flex>
+          <div slot="left">
+            <Flex align="left">
```
Comment (rudolfs) [inline ui/DesignSystem/Component/Transaction/Accordion.svelte:134]: You could avoid wrapping the content within another `<Flex align="left">
 ` by setting the parent `div`'s inline style to: `style="display: flex"`.

File: ui/DesignSystem/Component/Transaction/Accordion.svelte [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+              style="vertical-align: middle; margin-right: 16px;" />
+          </div>
+        </Flex>
+      </div>
+    {/each}
+  </div>
+  <div class="summary" on:click={hideCards}>
+    <Flex>
+      <div slot="left">
+        <Flex align="left">
```
Comment (rudolfs) [inline ui/DesignSystem/Component/Transaction/Accordion.svelte:160]: You could avoid wrapping the content within another <Flex align="left"> by setting the parent div's inline style to: style="display: flex".

File: ui/Screen/DesignSystemGuide.svelte [hunk 5]
```diff
@@ -636,6 +674,41 @@
     </Swatch>
   </Section>
 
+  <Section
```
Comment (rudolfs) [inline ui/Screen/DesignSystemGuide.svelte:677]: So far we've grouped all similar components under one section (and use `<Section>` tags to divide components into subgroups). Should we perhaps also do it for transactions for consistency's sake?

File: ui/Screen/TransactionDetails.svelte [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+  const formatMessage = {
+    USER_REGISTRATION: "User registration",
+    PROJECT_REGISTRATION: "Project registration"
+  };
+
+  const formatSubject = msg => {
+    return {
+      name:
+        msg.kind === "USER_REGISTRATION"
```
Comment (rudolfs) [inline ui/Screen/TransactionDetails.svelte:71]: This should also be a constant that we import.

File: ui/Screen/TransactionDetails.svelte [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+  const transactions = query(client, {
+    query: GET_TRANSACTIONS,
+    variables: {
+      ids: [params.id]
+    },
+    fetchPolicy: "no-cache"
+  });
+
+  const formatMessage = {
+    USER_REGISTRATION: "User registration",
```
Comment (rudolfs) [inline ui/Screen/TransactionDetails.svelte:64]: We should keep types coming from the proxy in one place.

So far we've kept them as constants (have a look how BLOB and TREE are used in components):
https://github.com/radicle-dev/radicle-upstream/blob/master/native/types.js

File: ui/Screen/DesignSystemGuide.svelte (hunk 5 above)
Comment (MeBrei) [reply ui/Screen/DesignSystemGuide.svelte:677]: It is grouped under the headline `Components`. Do you mean to have an additional `Transaction` group outside of `Components`?

**Comment by MeBrei** [comment]
> Additional thoughts:
> 
> * it seems to me that the `<Transaction>` component is trying to do too many things, maybe we can do a little pairing session to check if it could be refactored?

I agree, but leave it as is for now. When we implement Orgs and integrate wallets, we will have to touch the component again and it will get more complex. That will be a good time to refactor it.

> * this would be a good place to start doing unit or snapshot tests for single components, see: #178

I'd prefer to try to wrap up this PR. I could add a comment in that issue that transaction component tests could be part of that issue.

> * the nested ternary operators are a bit hard to parse but the only alternative which preserves reactivity would be using `&&` and `||`, which is not any better: [`IdentityAvatar.svelte`](https://github.com/radicle-dev/radicle-upstream/blob/master/ui/DesignSystem/Component/IdentityAvatar.svelte#L14:L17)

I agree that they are a bit clumsy right now. As a follow up to this PR I want to work on centralizing transaction handling and working on the reactivity of the components. I expect to have to touch many of those places then and will try to rework them more readable.

File: ui/Screen/DesignSystemGuide.svelte (hunk 5 above)
Comment (rudolfs) [reply ui/Screen/DesignSystemGuide.svelte:677]: Yeah, something like this:
```
Components
  Avatars
  ...
  Transactions
    <Swatch><TransactionRow></Swatch>
    <Swatch><Accordion></Swatch>
    <Swatch><TransactionStatusbar></Swatch>
```
